pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -203,7 +206,23 @@ func init() {
 }
 
 // Initialize passes a Kubernetes clientBuilder interface to the cloud provider
-func (vs *VSphere) Initialize(clientBuilder controller.ControllerClientBuilder) {}
+func (vs *VSphere) Initialize(clientBuilder controller.ControllerClientBuilder) {
+	if vs.cfg == nil {
+		return
+	}
+
+	// Only on controller node it is required to register listeners.
+	// Register callbacks for node updates
+	client := clientBuilder.ClientOrDie("vSphere-cloud-provider")
```

Comment:
This will stop the controller manager if we cannot establish a connection to the API server. This introduces a dependency on API server that did not exist earlier. 

I would suggest to retry connecting to API server instead of exiting/stopping the controller manager.

Reply:
Nothing will be required Sandeep.
I confirmed with "Josh Horwitz" the one who suggested the solution.
He said, after every provided interval, cache will be updated.

"Watches will reconnect automatically.
Informers use list & watch. Every interval it will list everything and the watches continuously watch for events."

This is his reply.
I will test the API server going down though.